src/extension.ts

```diff
@@ -264,6 +264,50 @@ async function runJavaFile(uri: vscode.Uri, noDebug: boolean) {
         throw ex;
     }
 
+    const hasMainMethods: boolean = mainMethods.length > 0;
+    const canRunTests: boolean = await canDelegateToJavaTestRunner(uri);
+
+    if (!hasMainMethods && !canRunTests) {
+        const workspaceFolder: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(uri);
+        if (!workspaceFolder) {
+            vscode.window.showErrorMessage(`Failed to run the project because the file: ${uri.fsPath} does not belongs to the current workspace.`);
+            return;
+        }
+        const mainClasses: IMainClassOption[] = await utility.searchMainMethods(workspaceFolder.uri);
```

Comment:
Can we just search the main method from the whole workspace directly instead of limiting to the belonging root?

Reply:
Update the utility method to make it possible to search all main classes in the workspace. Another question, do you think we need to differentiate the "mains" by appending the belonging project name in the list?


![image](https://user-images.githubusercontent.com/6193897/97655724-b47b4880-1aa0-11eb-8591-6334e3099548.png)

Reply:
yes. if it falls back to run the main from the workspace, adding project name in the list would help.

Reply:
New look:

![WeChat Screenshot_20201030135941](https://user-images.githubusercontent.com/6193897/97666360-6d994d00-1ab8-11eb-8b70-dfa5d39cade9.png)